modules/gapi/perf/common/gapi_core_perf_tests_inl.hpp

```diff
@@ -323,17 +323,23 @@ PERF_TEST_P_(DivPerfTest, TestPerformance)
     Size sz = get<1>(GetParam());
     MatType type = get<2>(GetParam());
     int dtype = get<3>(GetParam());
-    cv::GCompileArgs compile_args = get<4>(GetParam());
+    double scale = get<4>(GetParam());
+    cv::GCompileArgs compile_args = get<5>(GetParam());
 
     // FIXIT Unstable input data for divide
     initMatsRandU(type, sz, dtype, false);
 
+    //This condition need to workaround bug in OpenCV.
+    //It reinitializes divider matrix without zero values.
+    if (dtype == CV_16S && type != CV_16S)
+        cv::randu(in_mat2, cv::Scalar::all(1), cv::Scalar::all(255));
```

Comment:
BTW, There is similar problem in `cv::divide()` with dst=8S (e.g., src=8U).

Probably accurate condition should be `dtype != type` ( ` && dtype != CV_32F`)

Reply:
Fortunately, fluid's Div kernel doesn't support 8S type at all.

For supported types:
Since bug observed only when `DST_Type=16S and SRC_Type != 16S` (both condition together), I think accurate condition is
`(dtype == 16S) && (dtype != type) `

It makes no sense to add this condition `dtype != CV_32F`, since the initMatsRandU() function still initializes the matrices with integer values. Please take a look [here](https://github.com/opencv/opencv/blob/58043dac0f3e24620ba3c2229506a9c029e22ff9/modules/gapi/test/common/gapi_tests_common.hpp#L217)